src/endpoint/s3/ops/s3_put_bucket_policy.js

```diff
@@ -23,10 +23,42 @@ async function put_bucket_policy(req) {
         }
         if (statement.action) statement.action = _.flatten([statement.action]);
         if (statement.resource) statement.resource = _.flatten([statement.resource]);
+        if (statement.condition) statement.condition = _parse_condition(statement.condition);
     }
     await req.object_sdk.put_bucket_policy({ name: req.params.bucket, policy });
 }
 
+function _parse_condition(conditions) {
+    // https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition_operators.html
+    // TODO: this is a subset of the full list of conditions. should add more condtions in the future
+    const supported_conditions = {
+        'stringequals': 'string_equals',
+        'stringnotequals': 'string_not_equals',
+        'null': 'null'
+    };
+    //https://docs.aws.amazon.com/AmazonS3/latest/userguide/list_amazons3.html#amazons3-policy-keys
+    //TODO: this is a subset of the full list of condition keys. should add more condtion keys in the future
+    const condition_key_map = {
+        's3:x-amz-server-side-encryption': 'server_side_encryption'
+    };
+    const parsed_condition = {};
+    for (const [condition, obj] of Object.entries(conditions)) {
+        if (!supported_conditions[condition]) {
```

Comment:
the reason I added the map was to convert the s3 format to the DB format (same as all the other maps), since we convert the json to lowercase at the beginning of the function all the conditions seems to be a single word (the s3 style is camel case, so StringNotEqual will be converted to stringnotequal while I want to convert it to string_not_equal). so I don't think string manipulations will work in this case. I don't want to remove the conversion to lower case since it might hurt other functionality of the bucket policy. maybe I can just leave the conditions as single words in the DB as well.
In get-bucket-policy we already return all the policy values in lower case anyway